core/lib/multivm/src/tracers/call_tracer/era_vm/mod.rs

```diff
@@ -0,0 +1,192 @@
+        let current = execution.current_context().unwrap();
+        // All calls from the actual users are mimic calls,
+        // so we need to check that the previous call was to the deployer.
+        // Actually it's a call of the constructor.
+        // And at this stage caller is user and callee is deployed contract.
+        let call_type = if let CallType::Call(far_call) = current_call.r#type {
+            if matches!(far_call, FarCallOpcode::Mimic) {
+                let previous_caller = execution
+                    .running_contexts
+                    .first()
+                    .map(|call| call.caller)
+                    // Actually it's safe to just unwrap here, because we have at least one call in the stack
+                    // But i want to be sure that we will not have any problems in the future
+                    .unwrap_or(current.caller);
+                if previous_caller == CONTRACT_DEPLOYER_ADDRESS {
+                    CallType::Create
+                } else {
+                    CallType::Call(far_call)
+                }
+            } else {
+                CallType::Call(far_call)
+            }
+        } else {
+            unreachable!()
+        };
+        let calldata = if current.heap_id == 0 || current.frame.gas_left.0 == 0 {
+            vec![]
+        } else {
+            let packed_abi = execution.get_register(1);
+            assert!(packed_abi.is_pointer);
+            let pointer = FatPointer::decode(packed_abi.value);
+            execution
+                .heaps
+                .get(pointer.page)
+                .unwrap()
+                .read_unaligned_from_pointer(&pointer)
+                .unwrap_or_default()
+        };
+
+        current_call.input = calldata;
+        current_call.r#type = call_type;
+        current_call.from = current.caller;
+        current_call.to = current.contract_address;
+        current_call.value = U256::from(current.context_u128);
+        current_call.gas = current.frame.gas_left.0 as u64;
+    }
+
+    fn save_output_era(
+        &mut self,
+        execution: &Execution,
+        ret_opcode: RetOpcode,
+        current_call: &mut Call,
+    ) {
+        let fat_data_pointer = execution.get_register(1);
+
+        // if `fat_data_pointer` is not a pointer then there is no output
+        let output = if fat_data_pointer.is_pointer {
+            let fat_data_pointer = FatPointer::decode(fat_data_pointer.value);
+            if fat_data_pointer.len == 0 && fat_data_pointer.offset == 0 {
+                Some(
+                    execution
+                        .heaps
+                        .get(fat_data_pointer.page)
+                        .unwrap()
+                        .read_unaligned_from_pointer(&fat_data_pointer)
+                        .unwrap(),
+                )
+            } else {
+                None
+            }
+        } else {
+            None
+        };
```

Comment:
Consider this change, I think it's clearer:
```suggestion
              let output = match fat_data_pointer.is_pointer {
            true => {
                let fat_data_pointer = FatPointer::decode(fat_data_pointer.value);
                match (fat_data_pointer.len, fat_data_pointer.offset) {
                    (0, 0) => execution
                        .heaps
                        .get(fat_data_pointer.page)
                        .and_then(|ptr| ptr.read_unaligned_from_pointer(&fat_data_pointer).ok()),
                    _ => None,
                }
            }
            _ => None,
        };
```